batch-oai-harvest/src/main/java/data/config/OaiHarvestJobConfig.java

```diff
@@ -27,46 +32,71 @@
-  @Value("${oaiharvest.parallelization.size}")
-  public int parallelization;
+        return new StepBuilder(IDENTIFIERS_HARVEST_STEP_NAME, jobRepository)
+                .<ExecutionRecordExternalIdentifier, ExecutionRecordExternalIdentifier>chunk(chunkSize, transactionManager)
+                .reader(oaiRepositoryIdentifiersHarvester)
+                .writer(oaiIdentifiersWriter)
+                .taskExecutor(oaiHarvestStepAsyncTaskExecutor)
+                .build();
+    }
 
-  @Bean
-  public Job oaiHarvestBatchJob(JobRepository jobRepository, Step oaiHarvestStep) {
-    LOGGER.info("Chunk size: {}, Parallelization size: {}", chunkSize, parallelization);
-    return new JobBuilder(BATCH_JOB, jobRepository)
-        .incrementer(new TimestampJobParametersIncrementer())
-        .start(oaiHarvestStep)
-        .build();
-  }
+    @Bean("oaiRecordsHarvestStep")
+    public Step oaiRecordsHarvestStep(
+            JobRepository jobRepository,
+            OaiIdentifiersReader oaiIdentifiersReader,
+            @Qualifier("transactionManager") PlatformTransactionManager transactionManager,
+            OaiRecordHarvester oaiRecordHarvester,
+            ExecutionRecordDTOItemWriter writer,
+            @Qualifier("oaiHarvestStepAsyncTaskExecutor") TaskExecutor oaiHarvestStepAsyncTaskExecutor) {
 
-  @Bean
-  public Step oaiHarvestStep(JobRepository jobRepository,
-      @Qualifier("transactionManager") PlatformTransactionManager transactionManager,
-      OaiHarvestItemReader oaiHarvestItemReader,
-      ExecutionRecordDTOItemWriter writer,
-      @Qualifier("oaiHarvestStepAsyncTaskExecutor") TaskExecutor oaiHarvestStepAsyncTaskExecutor) {
-    return new StepBuilder(STEP_NAME, jobRepository)
-        .<ExecutionRecordDTO, ExecutionRecordDTO>chunk(chunkSize, transactionManager)
-        .reader(oaiHarvestItemReader)
-        .writer(writer)
-        //TODO: 2024-01-31 - Update with a better parallelization reader and remove Step parallization.
-        .taskExecutor(oaiHarvestStepAsyncTaskExecutor)
-        .throttleLimit(parallelization)
-        .build();
-  }
+        return new StepBuilder(RECORDS_HARVEST_STEP_NAME, jobRepository)
+                .<ExecutionRecordExternalIdentifier, ExecutionRecordDTO>chunk(chunkSize, transactionManager)
+                .reader(oaiIdentifiersReader)
+                .processor(oaiRecordHarvester)
+                .writer(writer)
+                .taskExecutor(oaiHarvestStepAsyncTaskExecutor)
+                .build();
+    }
 
-  @Bean
-  public TaskExecutor oaiHarvestStepAsyncTaskExecutor() {
-    ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
-    executor.setCorePoolSize(parallelization);
-    executor.setMaxPoolSize(parallelization);
-    executor.initialize();
-    return executor;
-  }
+    @Bean
+    public TaskExecutor oaiHarvestStepAsyncTaskExecutor() {
+        ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+        executor.setThreadNamePrefix("oaiThread-");
```

Comment:
I wouldn't extract it yet to another method. It would be too nested, this method here is just setting parameters. I think it would be unnecessary to nest this further.